engine/access/rpc/backend/backend_scripts.go

```diff
@@ -340,37 +363,48 @@ func (b *backendScripts) compareScriptExecutionResults(
 	}
 
 	b.metrics.ScriptExecutionResultMismatch()
-	b.logScriptExecutionComparison(execNodeResult, execErr, localResult, localErr, blockID, script, insecureScriptHash,
+	b.logScriptExecutionComparison(execNodeResult, execDuration, execErr, localResult, localDuration, localErr, blockID, script, arguments, insecureScriptHash,
 		"script execution results from local execution do not match EN")
 }
 
 // logScriptExecutionComparison logs the script execution comparison between local execution and execution node
 func (b *backendScripts) logScriptExecutionComparison(
 	execNodeResult []byte,
+	execDuration time.Duration,
 	execErr error,
 	localResult []byte,
+	localDuration time.Duration,
 	localErr error,
 	blockID flow.Identifier,
 	script []byte,
+	arguments [][]byte,
 	insecureScriptHash [md5.Size]byte,
 	msg string,
 ) {
+	args := make([]string, len(arguments))
+	for i, arg := range arguments {
+		args[i] = string(arg)
+	}
+
 	lgCtx := b.log.With().
 		Hex("block_id", blockID[:]).
+		Hex("script_hash", insecureScriptHash[:]).
 		Str("script", string(script)).
-		Hex("script_hash", insecureScriptHash[:])
+		Strs("args", args)
```

Comment:
is there a case where args might contain some confidential information or secret values? In case you have your own AN deployment and you might not be aware things are getting logged. Just thinking out loud.